From da5cbd1a3b248f2d32281a1766a3d1414c0e8e03 Mon Sep 17 00:00:00 2001
From: Sylvain Gault <sylvain.gault@gmail.com>
Date: Tue, 29 Sep 2015 02:38:25 +0200
Subject: [PATCH] bios: Fix alignment change with gcc 5

The section aligment specified in the ld scripts have to be greater or
equal to those in the .o files generated by gcc.

Signed-off-by: Sylvain Gault <sylvain.gault@gmail.com>
Tested-by: poma <pomidorabelisima@gmail.com>
Signed-off-by: Paulo Alcantara <pcacjr@zytor.com>
Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 core/i386/syslinux.ld   | 6 +++---
 core/x86_64/syslinux.ld | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/core/i386/syslinux.ld b/core/i386/syslinux.ld
index 7b4e012..7390451 100644
--- a/core/i386/syslinux.ld
+++ b/core/i386/syslinux.ld
@@ -266,7 +266,7 @@ SECTIONS
 		__text_end = .;
 	}
 
-	. = ALIGN(16);
+	. = ALIGN(32);
 
 	__rodata_vma = .;
 	__rodata_lma = __rodata_vma + __text_lma - __text_vma;
@@ -361,7 +361,7 @@ SECTIONS
 		__dynamic_end = .;
 	}
 
-	. = ALIGN(16);
+	. = ALIGN(32);
 
 	__data_vma = .;
 	__data_lma = __data_vma + __text_lma - __text_vma;
@@ -377,7 +377,7 @@ SECTIONS
 	__pm_code_dwords = (__pm_code_len + 3) >> 2;
 
 	. = ALIGN(128);
-	
+
 	__bss_vma = .;
 	__bss_lma = .;		/* Dummy */
 	.bss (NOLOAD) : AT (__bss_lma) {
diff --git a/core/x86_64/syslinux.ld b/core/x86_64/syslinux.ld
index 1057112..bf815c4 100644
--- a/core/x86_64/syslinux.ld
+++ b/core/x86_64/syslinux.ld
@@ -266,7 +266,7 @@ SECTIONS
 		__text_end = .;
 	}
 
-	. = ALIGN(16);
+	. = ALIGN(32);
 
 	__rodata_vma = .;
 	__rodata_lma = __rodata_vma + __text_lma - __text_vma;
@@ -361,7 +361,7 @@ SECTIONS
 		__dynamic_end = .;
 	}
 
-	. = ALIGN(16);
+	. = ALIGN(32);
 
 	__data_vma = .;
 	__data_lma = __data_vma + __text_lma - __text_vma;
@@ -377,7 +377,7 @@ SECTIONS
 	__pm_code_dwords = (__pm_code_len + 3) >> 2;
 
 	. = ALIGN(128);
-	
+
 	__bss_vma = .;
 	__bss_lma = .;		/* Dummy */
 	.bss (NOLOAD) : AT (__bss_lma) {
-- 
2.7.4

